iac-extensions/arm/src/test/java/org/sonar/iac/arm/tree/impl/bicep/ArrayExpressionImplTest.java

```diff
@@ -47,13 +52,33 @@ void shouldParseMultilineArrays() {
 
   @Test
   void shouldParseValidExpression() {
-    ArrayExpression tree = (ArrayExpression) createParser(BicepLexicalGrammar.ARRAY_EXPRESSION).parse(
-      code("[", "'a'", "'b'", "]"));
+    ArrayExpression tree = parse(code("[", "'a'", "'b'", "]"), BicepLexicalGrammar.ARRAY_EXPRESSION);
 
     SoftAssertions softly = new SoftAssertions();
     softly.assertThat(tree).isInstanceOf(ArrayExpression.class);
     softly.assertThat(tree.getKind()).isEqualTo(ArmTree.Kind.ARRAY_EXPRESSION);
     softly.assertThat(tree.elements()).hasSize(2);
     softly.assertAll();
   }
+
+  @Test
+  void shouldParseMixedInlineAndMultilineArray() {
+    String code = code("[",
+      "'a', 'b'",
+      "'c'",
+      "]");
+    ArrayExpression tree = parse(code, BicepLexicalGrammar.ARRAY_EXPRESSION);
+
+    assertThat(tree.getKind()).isEqualTo(ArmTree.Kind.ARRAY_EXPRESSION);
+    assertThat(tree.elements()).hasSize(3);
+    StringComplete a = (StringComplete) tree.elements().get(0);
+    StringComplete b = (StringComplete) tree.elements().get(1);
+    StringComplete c = (StringComplete) tree.elements().get(2);
+    ArmAssertions.assertThat(a.content()).hasValue("a");
+    ArmAssertions.assertThat(b.content()).hasValue("b");
+    ArmAssertions.assertThat(c.content()).hasValue("c");
+
```

Comment:
please add assertions for `separators()`

Reply:
Not possible here: `ArrayExpression` does not provide access to the `SeparatedList` object.
I added another test in `SeparatedListImplTest` about it.